azurerm/internal/services/sentinel/resource_arm_sentinel_alert_rule_action.go

```diff
@@ -0,0 +1,179 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	logicParse "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/logic/parse"
+	logicValidate "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/logic/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/sentinel/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/sentinel/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+const OperationInsightsRPName = "Microsoft.OperationalInsights"
+
+func resourceArmSentinelAlertRuleAction() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmSentinelAlertRuleActionCreate,
+		Read:   resourceArmSentinelAlertRuleActionRead,
+		Delete: resourceArmSentinelAlertRuleActionDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.SentinelAlertRuleActionID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"rule_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.SentinelAlertRuleID,
+			},
+
+			"logic_app_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: logicValidate.LogicAppWorkflowID,
+			},
+
+			"logic_app_trigger_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceArmSentinelAlertRuleActionCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Sentinel.AlertRulesClient
+	logicTriggerClient := meta.(*clients.Client).Logic.WorkflowTriggersClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+
+	ruleID, err := parse.SentinelAlertRuleID(d.Get("rule_id").(string))
+	if err != nil {
+		return err
+	}
+	lappId, err := logicParse.LogicAppWorkflowID(d.Get("logic_app_id").(string))
+	if err != nil {
+		return err
+	}
+
+	// Ensure no existed resources
+	resp, err := client.GetAction(ctx, ruleID.ResourceGroup, OperationInsightsRPName, ruleID.Workspace, ruleID.Name, name)
```

Comment:
isn't threading the RP through here a swagger bug?

Reply:
I'm sorry to be dumb here, what do you mean is a swagger bug?